src/Knight/Knight.cpp

```diff
@@ -0,0 +1,79 @@
+#include<bits/stdc++.h>
+
+using namespace std;
+
+// 8 vi tri chu L ma con ma co the di
+int dx[8] = {2, 1, -1, -2, -2, -1, 1, 2};
+int dy[8] = {1, 2, 2, 1, -1, -2, -2, -1};
+
+// kiem tra xem buoc di cua con ma co hop le hay khong ?
+bool Checkmove(int x, int y, int n, const vector<vector<int>>& board) {
+    return (x >= 0 && y >= 0 && x < n && y < n && board[x][y] == -1);
+}
+
+// backtracking
+bool Hanhtrinh(int u, int v, int moveCount, vector<vector<int>>& board, int n) {
+	// kiem tra xem con ma da di du so buoc hay chua, toi da n*n buoc
+    if (moveCount == n * n) {
+        return true;
+    }
+	//cho con ma di chuyen theo chu L
+    for (int i = 0; i < 8; i++) {
+        int nextu = u + dx[i];
+        int nextv = v + dy[i];
+        // neu duoc thi di tiep cho den khi khong the di duoc nua
+        if (Checkmove(nextu, nextv, n, board)) {
+            board[nextu][nextv] = moveCount;
+            if (Hanhtrinh(nextu, nextv, moveCount + 1, board, n)) {
+                return true;
+            }
+            board[nextu][nextv] = -1;
+        }
+    }
+    return false;
+}
+
+// Ham sap sep lai vector 2 chieu
+void sortBoard(vector<vector<int>>& board) {
```

Comment:
why we have to sort the whole board?

Reply:
Because, I want to show the move count sequentially. For example 
1:(0,0) 
2:(4,2) 
3:(3,4) 
4: ....
 But It seems failed. Thank you for reminding me.  